hermes/src/store/types.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    super::proof::wormhole_merkle::WormholeMerkleMessageProof,
+    pyth_oracle::{
+        Message,
+        MessageType,
+        PriceFeedMessage,
+    },
+};
+
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct MessageIdentifier {
+    pub price_id: [u8; 32],
+    pub type_:    MessageType,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct ProofSet {
+    pub wormhole_merkle_proof: WormholeMerkleMessageProof,
+}
+
+pub type Slot = u64;
+pub type UnixTimestamp = i64;
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum RequestTime {
+    Latest,
+    FirstAfter(UnixTimestamp),
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct AccumulatorMessages {
+    pub magic:     [u8; 4],
+    pub slot:      Slot,
+    pub ring_size: u32,
+    pub messages:  Vec<Message>,
+}
+
+impl AccumulatorMessages {
+    pub fn ring_index(&self) -> u32 {
+        (self.slot % self.ring_size as u64) as u32
+    }
+}
+
+pub enum Update {
+    Vaa(Vec<u8>),
+    AccumulatorMessages(AccumulatorMessages),
+}
+
+pub struct PriceFeedUpdate {
```

Comment:
this type and the one below seem like they should go in the API layer (because they're an implementation detail of the current REST API, and will go away in the future). I think you probably want to move `Store.get_price_feeds_with_update_data` to the API layer as well.